chia/wallet/trading/trade_store.py

```diff
@@ -133,6 +134,12 @@ async def create(
             except aiosqlite.OperationalError:
                 pass  # ignore what is likely Duplicate column error
 
+            try:
+                await conn.execute("ALTER TABLE trade_records ADD COLUMN offer_name blob")
+                await conn.execute("CREATE INDEX IF NOT EXISTS trade_offer_name on trade_records(offer_name)")
+            except aiosqlite.OperationalError:
+                pass  # ignore what is likely Duplicate column error
+
```

Comment:
offer_name is already in the DB as trade_id, it's the primary key.  The root issue you're trying to get at here I think is not solved by adding this because if you take an offer it will use some of your coins and make a complimentary offer and save that trade record.  Then, if you try to take that same offer again, you will use new coins since your first coins are locked up in the first trade record.  That will result in a different name and you will still have a "duplicate" take of the same offer in the DB.

Reply:
there's a difference between completed offer and offer taken. Offer taken is the one that needs to be deduped.